engineapi/proxy_replica.go

```diff
@@ -4,10 +4,10 @@ import (
 	longhorn "github.com/longhorn/longhorn-manager/k8s/pkg/apis/longhorn/v1beta2"
 )
 
-func (p *Proxy) ReplicaAdd(e *longhorn.Engine, replicaName, replicaAddress string, restore, fastSync bool, replicaFileSyncHTTPClientTimeout int64) (err error) {
+func (p *Proxy) ReplicaAdd(e *longhorn.Engine, replicaName, replicaAddress string, restore, fastSync bool, replicaFileSyncHTTPClientTimeout, grpcTimeoutSeconds int64) (err error) {
 	return p.grpcClient.ReplicaAdd(string(e.Spec.DataEngine), e.Name, e.Spec.VolumeName, p.DirectToURL(e),
 		replicaName, replicaAddress, restore, e.Spec.VolumeSize, e.Status.CurrentSize,
-		int(replicaFileSyncHTTPClientTimeout), fastSync)
+		int(replicaFileSyncHTTPClientTimeout), fastSync, grpcTimeoutSeconds)
```

Comment:
Should we handle the compatibility of talking to an old instance manager?

Reply:
I think it should work since the old IM will just ignore the new gRPC field. WDYT?

Reply:
I agree with @PhanLe1010. According to https://kubernetes.io/blog/2023/08/25/native-sidecar-containers/#how-did-users-get-sidecar-behavior-before-1-28 (and my experience), it is fine to send the new field to the old service. It will simply be ignored.

For this change, we also have the following scenario if the old instance-manager communicates with a new engine (running inside it) or replica (running elsewhere):

- Old instance-manager ignores `grpcTimeoutSeconds` from longhorn-manager.
- Old instance-manager does not send `grpcTimeoutSeconds` to new engine or replica.
- New engine or replica uses default value (0).

IIRC, this scenario is also fine, since @PhanLe1010's updates handle the default value by using the old `grpcLongTimeout`. Correct me if I'm wrong though.

Reply:
> IIRC, this scenario is also fine, since @PhanLe1010's updates handle the default value by using the old grpcLongTimeout. Correct me if I'm wrong though.

Yeah, you are correct